synapse/lib/storm.py

```diff
@@ -1946,6 +1947,138 @@ async def execStormCmd(self, runt, genr):
 
         await runt.printf('For detailed help on any command, use <cmd> --help')
 
+class MergeCmd(Cmd):
+    '''
+    Merge edits from the incoming nodes down to the next layer.
+
+    NOTE: This command requires the current view to be a fork.
+
+    Examples:
+
+        // Having tagged a new #cno.mal.redtree subgraph in a forked view...
+
+        #cno.mal.redtree | merge --apply
+
+        // Print out what the merge command *would* do but dont.
+
+        #cno.mal.redtree | merge
+
+    '''
+    name = 'merge'
+
+    def getArgParser(self):
+        pars = Cmd.getArgParser(self)
+        pars.add_argument('--apply', default=False, action='store_true', help='Execute the merge changes.')
+        return pars
+
+    async def execStormCmd(self, runt, genr):
+
+        if runt.snap.view.parent is None:
+            mesg = 'You may only merge nodes in forked views'
+            raise s_exc.CantMergeView(mesg=mesg)
+
+        layr0 = runt.snap.view.layers[0].iden
+        layr1 = runt.snap.view.layers[1].iden
+
+        async for node, path in genr:
+
+            # the timestamp for the adds/subs of each node merge will match
+            nodeiden = node.iden()
+            meta = {'user': runt.user.iden, 'time': s_common.now()}
+
+            sode = node.sodes[-1]
+
+            adds = []
+            subs = []
+
+            async def sync():
+
+                if not self.opts.apply:
+                    adds.clear()
+                    subs.clear()
+                    return
+
+                if adds:
+                    addedits = [(node.buid, node.form.name, adds)]
+                    await runt.snap.view.layers[1].storNodeEdits(addedits, meta=meta)
+                    adds.clear()
+
+                if subs:
+                    subedits = [(node.buid, node.form.name, subs)]
+                    await runt.snap.view.layers[0].storNodeEdits(subedits, meta=meta)
+                    subs.clear()
+
+            # check all node perms first
+            form = sode.get('form')
+            valu = sode.get('valu')
+            if valu is not None:
+                runt.confirm(('node', 'add', form), gateiden=layr0)
+                adds.append((s_layer.EDIT_NODE_ADD, valu, ()))
+                subs.append((s_layer.EDIT_NODE_DEL, valu, ()))
```

Comment:
Doesn't seem like we actually need to append the edits to `adds` and `subs` if they aren't going to be applied.

Reply:
I left this to be as close to a "dry run" as possible, probably not worth it :D